File: contests/lc/2211.CountCollisionsonaRoad/solution.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        """
        Trim outward-moving cars (leading L's, trailing R's); all remaining
        moving cars will collide and become stationary.
        """
        n = len(directions)

        left = 0
        while left < n and directions[left] == "L":
            left += 1

        right = n - 1
        while right >= left and directions[right] == "R":
            right -= 1

        collisions = 0
        for i in range(left, right + 1):
            if directions[i] != "S":
                collisions += 1

        return collisions

File: contests/lc/2211.CountCollisionsonaRoad/solution.go
package main

// countCollisions returns the total number of collisions on the road.
// Cars that forever move outward (leading L's, trailing R's) never collide,
// so we trim them and count every remaining moving car (non-'S') as a collision.
func countCollisions(directions string) int {
	bytes := []byte(directions)
	n := len(bytes)

	left := 0
	for left < n && bytes[left] == 'L' {
		left++
	}

	right := n - 1
	for right >= left && bytes[right] == 'R' {
		right--
	}

	collisions := 0
	for i := left; i <= right; i++ {
		if bytes[i] != 'S' {
			collisions++
		}
	}

	return collisions
}
